analytic_engine/src/sst/parquet/encoding.rs

```diff
@@ -150,3 +173,93 @@ pub fn decode_sst_meta_data(kv: &KeyValue) -> Result<SstMetaData> {
 
     SstMetaData::try_from(meta_data_pb).context(ConvertSstMetaData)
 }
+
+pub struct ParquetEncoder<W: Write> {
+    writer: ArrowWriter<W>,
+    format: StorageFormat,
+    origin_arrow_schema: ArrowSchemaRef,
+    hybrid_arrow_schema: Option<ArrowSchemaRef>,
+    schema: Schema,
+}
+
+impl<W: Write> ParquetEncoder<W> {
+    pub fn try_new(
```

Comment:
should we check if the format is compatible with the table? E.g., some var-len types are not supported and we can check in advance here.